pints/_log_likelihoods.py

```diff
@@ -861,6 +862,323 @@ def evaluateS1(self, x):
+        super(CensoredGaussianLogLikelihood, self).__init__(problem)
+
+        # Get number of times, number of outputs
+        self._nt = len(self._times)
+        self._no = problem.n_outputs()
+
+        # Add parameters to problem
+        self._n_parameters = problem.n_parameters() + self._no
+        a = lower
+        if a is not None:
+            if np.isscalar(a):
+                a = np.ones(self._no) * float(a)
+            else:
+                a = pints.vector(a)
+                if len(a) != self._no:
+                    raise ValueError(
+                        'Lower limit must be a ' +
+                        ' scalar or a vector of length n_outputs.')
+        b = upper
+        if b is not None:
+            if np.isscalar(b):
+                b = np.ones(self._no) * float(b)
+            else:
+                b = pints.vector(b)
+                if len(b) != self._no:
+                    raise ValueError(
+                        'Upper limit must be a ' +
+                        ' scalar or a vector of length n_outputs.')
+
+        if (a is not None) and (b is not None):
+            diff = b - a
+            if np.any(diff <= 0):
+                raise ValueError('Upper limit ' +
+                                 'must exceed lower limit.')
+        self._a = a
+        self._b = b
+
+        # Define the condition for whether a point is not censored
+        if (self._a is not None) and (self._b is not None):
+            self._condition = (self._a < self._values
+                               ) & (self._values < self._b)
+        elif (self._a is not None) and (self._b is None):
+            self._condition = self._a < self._values
+        elif (self._a is None) and (self._b is not None):
+            self._condition = self._values < self._b
+        else:
+            raise ValueError("Not a censored likelihood")
+
+        # Check that none of the values are less than the lower
+        # limit or greater than the upper limit
+        if self._values.ndim == 1:
+            for i in range(np.shape(self._values)[0]):
+                if a is not None:
+                    if self._values[i] < self._a[0]:
+                        raise ValueError("A value is below the lower limit.")
+                if b is not None:
+                    if self._values[i] > self._b[0]:
+                        raise ValueError("A value is above the upper limit.")
+        else:
+            for i in range(np.shape(self._values)[1]):
+                for j in range(np.shape(self._values)[0]):
+                    if a is not None:
+                        if self._values[j, i] < self._a[i]:
+                            raise ValueError(
+                                "A value is below the lower limit.")
+                    if b is not None:
+                        if self._values[j, i] > self._b[i]:
+                            raise ValueError(
+                                "A value is above the upper limit.")
```

Comment:
Content comment:

Perhaps, I misunderstood the use of the likelihood. But isn't the purpose of the sensored likelihood that measurement values below the threshold exist but the measurement values can not be trusted because the measuring device has reached its resolution limit or equivalent? 

I think you could do without this check.

Reply:
I’ve removed this check. I’ve seen values at the limit of detection recorded as equal to it, but I think you’re correct that in general this may not be the case.